Plutarch/DataRepr/Internal.hs

```diff
@@ -121,6 +124,35 @@ instance PlutusType (PDataRecord '[]) where
   pcon' PDNil = pnil
   pmatch' _ f = f PDNil
 
+-- | This uses data equality. 'PEq' instances of elements don't make any difference.
+instance PEq (PDataRecord xs) where
+  x #== y = pdToBuiltin x #== pdToBuiltin y
+
+-- Lexicographic ordering based 'Ord' instances for 'PDataRecord'.
+
+instance POrd (PDataRecord '[]) where
+  _ #<= _ = pconstant True
+  _ #< _ = pconstant False
+
+instance (POrd x, PIsData x) => POrd (PDataRecord '[label ':= x]) where
+  l1 #< l2 = unTermCont $ do
+    PDCons x _ <- tcont $ pmatch l1
+    PDCons y _ <- tcont $ pmatch l2
+
+    pure $ pfromData x #< pfromData y
+  l1 #<= l2 = pdToBuiltin l1 #== pdToBuiltin l1 #|| l1 #< l2
```

Comment:
I have yet to test whether this pattern of "`#==` or `#<`" is better for `#<=` impl or if having a separate explicit definition similar to `#<=` is better.

Reply:
An explicit implementation here can be more efficient.

Reply:
I'll give explicit impls a try and see how that fares. Do you think explicit impls are necessary for only `PDataRecord`s or could it be beneficial at the `PDataSum` level too?

The interesting thing to think about is that the current implementation has a happy path when things are equal - this won't be the case for an iterative, explicit impl. But it could still liberate perf in non-empty case.